src/core/services/IronBankService.ts

```diff
@@ -94,31 +94,48 @@ export class IronBankServiceImpl implements IronBankService {
     const { MAX_UINT256 } = this.config;
     const provider = this.web3Provider.getSigner();
     const ironBankMarketContract = getContract(marketAddress, ironBankMarketAbi, provider);
-
     switch (action) {
       case 'supply':
-        return await this.transactionService.execute({ network, fn: ironBankMarketContract.mint, args: [amount] });
+        return await this.transactionService.execute({
+          network,
+          methodName: 'mint',
+          abi: ironBankMarketAbi,
+          contractAddress: marketAddress,
+          args: [amount],
+        });
       case 'withdraw':
         if (amount === MAX_UINT256) {
           const balanceOf = await ironBankMarketContract.balanceOf(userAddress);
```

Comment:
Isn't userAddress derivable from the web3Provider directly instead of having to ask it for every call?